python/ray/includes/libcoreworker.pxd

```diff
@@ -100,7 +100,7 @@ cdef extern from "ray/core_worker/core_worker.h" nogil:
         int MaxPendingCalls() const
 
     cdef cppclass CCoreWorker "ray::core::CoreWorker":
-        void ConnectToRaylet()
+        void ConnectToRaylet() except +
```

Comment:
this is required by Cython since the connection can now throw a runtime error

Reply:
This is supposed to tie back to the error thrown in `raylet_client.cc`, right? Maybe I'm misreading this,  but as far as I can see:
- You added a runtime error in `raylet::RayletConnection::RayletConnection` at raylet_client.cc L57
- This connection is setup in `raylet::RayletClient::RayletClient` at raylet_client.cc L107
- `CoreWorker::CoreWorker` creates a rayletclient at core_worker.cc L111
- `CoreWorker::ConnectToRaylet` then uses that rayletclient to do `AnnounceWorkerPort` , but so far as I can tell, the client and its connection are already created?

... so I'm not confident,if your runtime error is thrown below, won't it happen before a call to `ConnectToRaylet` can be made?

Reply:
Makes sense, I followed our previous patch for 0.8.4 but it's not applicable here as you pointed out.
- in pythonland `Worker.connect` call the constructor`ray._raylet.CoreWorker` (`python/ray/worker.py:1534`)
- in `python/ray/_raylet.pyx` at the end of the constructor `_raylet.CoreWorker` it calls `CCoreWorkerProcess.Initialize(options)`
- this function maps to `ray::core::CoreWorkerProcess::Initialize` (as defined in `python/ray/includes/libcoreworker.pxd`) which makes a call to `new CoreWorkerProcessImpl(options)`
- This method in turns calls `CreateWorker();` (see `src/ray/core_worker/core_worker_process.cc`)
- `CreateWorker` calls the constructor `CoreWorker::CoreWorker`

Hence, I believe the `except +` declaration needs to go in `libcoreworker.pxd`:
```
    cdef cppclass CCoreWorkerProcess "ray::core::CoreWorkerProcess":
        @staticmethod
        void Initialize(const CCoreWorkerOptions &options) except +
```